File: modulo-arithmetic/ModuloArithmetic.js
class ModuloArithmetic {
  static add(a, b, mod) {
    return ( (a % mod) + (b % mod) ) % mod;
  }

  static subtract(a, b, mod) {
    return ( (a % mod) - (b % mod) + mod ) % mod;
  }

  static multiply(a, b, mod) {
    return ( (a % mod) * (b % mod) ) % mod;
  }

  static power(base, exponent, mod) {
    let result = 1;
    base = base % mod;
    while (exponent > 0) {
      if (exponent % 2 == 1) {
        result = (result * base) % mod;
      }
      exponent = exponent >> 1; // exponent /= 2
      base = (base * base) % mod;
    }
    return result;
  }

  static modInverse(a, mod){
    return this.power(a, mod - 2, mod);
  }

  static divide(a, b, mod){
    return (a % mod * this.modInverse(b, mod) % mod) % mod;
  }
}

File: modulo-arithmetic/ModuloArithmetic.cs
class ModuloArithmetic
{
    public static long Add(long a, long b, long mod){
        return ((a % mod) + (b % mod)) % mod;
    }
    public static long Subtract(long a, long b, long mod){
        return ((a % mod) - (b % mod) + mod) % mod;
    }
    public static long Multiply(long a, long b, long mod){
        return ((a % mod) * (b % mod)) % mod;
    }
    public static long Power(long a, long b, long mod){
        long result = 1;
        a = a % mod;
        while(b > 0){
            if((b & 1) == 1){
                result = (result * a) % mod;
            }
            a = (a * a) % mod;
            b = b >> 1;
        }
        return result;
    }
    public static long ModInverse(long a, long mod){
        return Power(a, mod - 2, mod);
    }
    public static long Divide(long a, long b, long mod){
        return (a % mod * ModInverse(b, mod) % mod) % mod;
    }
}